app/src/androidTest/java/org/mozilla/focus/activity/ChangeLanguageTest.java

```diff
@@ -0,0 +1,127 @@
+package org.mozilla.focus.activity;
+
+
+import android.content.Intent;
+import android.content.res.Resources;
+import android.preference.Preference;
+import android.support.annotation.NonNull;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.uiautomator.UiDevice;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiScrollable;
+import android.support.test.uiautomator.UiSelector;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mozilla.focus.R;
+import org.mozilla.focus.helper.BeforeTestTask;
+import org.mozilla.focus.utils.AndroidTestUtils;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.PreferenceMatchers.withKey;
+import static android.support.test.espresso.matcher.RootMatchers.isDialog;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+public class ChangeLanguageTest {
+
+    // This is the lang name of Android system. We hard code it here so it will only fits emulator
+    // For physical devices this may vary.
+    private final String INDOLAN = "Indonesia";
```

Comment:
I believe we can get this string from Locale class

Reply:
After discussion we add -> To do : we hard code the language may apply getPackagedLocaleTags(getContext()) later